FaceDetectionServer/REST-API.md

```diff
@@ -0,0 +1,240 @@
+
+### POST
+```
+http://<hostname>:8008/recognizer/predict/
+```
+
+#### Parameters
+| Parameter | Description |
+| - | - |
+| image | The face image. |
+
+#### Return
+JSON string with a "success" value of "true" or "false", and if successful, the following JSON values:
+
+| Attribute | Description |
+| - | - |
+| success | "True" if face recognized, otherwise "false". |
+| subject | Name of detected face. |
+| confidence | Level of confidence. Actually a measure of distance the features are from matching a given subject. Lower is better, with 0 being a perfect match. |
+| rect | Coordinates of detected face. |
+| server_processing_time | Time in milliseconds that it took the server to process this image. |
+
+_Note: This call will return subject names even with very poor (high distance) confidence values. It is up to the caller to decide what it considers a good prediction. For instance in the [MobiledgeX SDK Demo app](https://play.google.com/store/apps/details?id=com.mobiledgex.sdkdemo), the *RECOGNITION_CONFIDENCE_THRESHOLD* value is 105, which means that any image with a confidence value higher than that will be shown with a subject name of "Unknown"._
+
+#### Examples:
+##### Face detected and recognized:
+```
+curl facedetection.defaultedge.mobiledgex.net:8008/recognizer/predict/ -F "image=@Bruce.jpg"
+{"success": "true", "subject": "Bruce Armstrong", "confidence": "71.850", "server_processing_time": "48.218, "rect": [73, 78, 146, 151]}
+```
+
+##### Face not detected:
+```
+curl facedetection.defaultedge.mobiledgex.net:8008/recognizer/predict/ -F "image=@single_pixel.png"
+{"success": "false", "server_processing_time": "0.082"}
+```
+##### Training data update in progress:
+```
+curl facedetection.defaultedge.mobiledgex.net:8008/recognizer/predict/ -F "image=@Bruce.jpg"
+<HTTP/1.1 503 Service Unavailable>
+Training data update in progress
+```
+If this state is encountered, it should only last a few seconds.
+
+## Pose Detection
+`/openpose/detect`
+
+Used to send an image with one or more human bodies to the server and get back a set of coordinates for any detected poses. In the our implementation, pose detection requires a GPU in order to run. If the server executing this call does not have GPU support, a `501 Not Implemented` response will be returned along with the message "OpenPose not supported on this server".
+
+### POST
+```
+http://<hostname>:8008/openpose/detect/
+```
+
+#### Parameters
+| Parameter | Description |
+| - | - |
+| image | The body/bodies image. |
+
+#### Return
+JSON string with the following attributes.
+
+| Attribute | Description |
+| - | - |
+| success | "True" if pose detected, otherwise "false". |
+| server_processing_time | The time in milliseconds that it took the server to process this image. |
+| poses | A JSON array of coordinates containing the detected poses. Only included if "success"="true". |
```

Comment:
and confidence values. Would be helpful to define the confidence value like was done for face recognition assuming it is the same